File: week 6/pset/mario/less/mario.py
def main():
    # Prompts user for an input between 1 and 8 inclusive
    while True:
        try:
            height = int(input("Height: "))
            if height >= 1 and height <= 8:
                break
        # If the input is not an integer prints an error message
        except ValueError:
            print("Invalid input")
    draw(height)
    
    
# Draws the pyramid          
def draw(height):
    for i in range(height):
        for j in range(height):
            if i + j >= height - 1:
                print("#", end="")
            else:
                print(" ", end="")
        print()


if __name__ == '__main__':
    main()

File: week 4/lecture/mario.c
#include <stdio.h>
#include <cs50.h>

void draw(int h);
int main(void)
{
    int height = get_int("Height: ");
    draw(height);
}

void draw(int h)
{
    if (h == 0)
    {
        return;
    }
    
    draw(h - 1);
    
    for (int i = 0; i < h; i++)
    {
        printf("#");
    }
    printf("\n");
}